Data/Generate.cpp

```diff
@@ -263,20 +274,19 @@ generate(unsigned int NumberOfEvents, const ComPWA::Kinematics &Kinematics,
     if (CurrentStartIndex >= LastIndex)
       break;
   }
-  double gen_eff = (double)events.size() / NumberOfEvents;
+  double gen_eff = (double)GeneratedEvents.Events.size() / NumberOfEvents;
   if (CurrentStartIndex > NumberOfEvents) {
-    gen_eff = (double)events.size() / CurrentStartIndex;
+    gen_eff = (double)GeneratedEvents.Events.size() / CurrentStartIndex;
   }
   LOG(INFO) << "Efficiency of toy MC generation: " << gen_eff;
 
-  return events;
+  return GeneratedEvents;
 }
 
-std::vector<ComPWA::Event>
-generatePhsp(unsigned int nEvents,
-             const ComPWA::PhaseSpaceEventGenerator &Generator,
-             ComPWA::UniformRealNumberGenerator &RandomGenerator) {
-  std::vector<ComPWA::Event> sample;
+EventCollection generatePhsp(unsigned int nEvents, std::vector<pid> Pids,
```

Comment:
Why can't the generator interface be changed to return EventCollections? I believe this would make things easier

Reply:
It does now, right?

Reply:
I mean this interface "Core/Generator.hpp"

Reply:
There is no mention of `std::vector<Event>` in `Core/Generator.hpp`

Reply:
Thats the problem. I believe it should be returning an EventCollection

Reply:
All functions in `Generator.hpp` are returning `EventCollection`s